app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala

```diff
@@ -302,10 +304,12 @@ class IFConnector @Inject()(httpClient: HttpClient, metrics: Metrics, agentCache
         ec)
     }
 
-  private def ifHeaders(authToken: String, env: String) = Seq(
+  def ifHeaders(authToken: String, env: String)(implicit hc: HeaderCarrier) = Seq(
     HeaderNames.authorisation -> s"Bearer $authToken",
     Environment               -> env,
-    CorrelationId             -> UUID.randomUUID().toString
+    CorrelationId             -> UUID.randomUUID().toString,
+    SessionId                 -> hc.sessionId.map(_.value).getOrElse(""),
```

Comment:
Do not know spec but why we add headers with empty strings ?

Reply:
The empty string was only as a default if no sessionId found. Might not be the best way to deal with it?

Reply:
I thiink when no session not found we should not send that header.